pkg/openai/main.go

```diff
@@ -0,0 +1,248 @@
+package openai
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/instill-ai/connector/pkg/base"
+	"github.com/instill-ai/connector/pkg/configLoader"
+
+	modelPB "github.com/instill-ai/protogen-go/model/model/v1alpha"
+	connectorPB "github.com/instill-ai/protogen-go/vdp/connector/v1alpha"
+)
+
+const (
+	venderName         = "openAI"
+	host               = "https://api.openai.com"
+	jsonMimeType       = "application/json"
+	reqTimeout         = time.Second * 60 * 5
+	textGenerationTask = "Text Generation"
+	textEmbeddingsTask = "Text Embeddings"
+)
+
+var (
+	//go:embed config/seed/definitions.json
+	definitionJSON []byte
+	once           sync.Once
+	connector      base.IConnector
+	taskToNameMap  = map[string]modelPB.Model_Task{
+		textGenerationTask: modelPB.Model_TASK_TEXT_GENERATION,
+		textEmbeddingsTask: modelPB.Model_TASK_TEXT_GENERATION, //to be changed?
```

Comment:
It is a good question. It is a little bit messed up that where should we put the task enums. I think we better not to use `modelPB` here anymore, because the instill-model should not be part of vdp. I'll think about it today.